packages/ui-components/src/components/modal/modal.tsx

```diff
@@ -0,0 +1,63 @@
+import { Component, Host, h, Prop, Event, EventEmitter } from '@stencil/core';
+import { EIconName } from '../icon/icon.types';
+import { IModalConfig, IModalEvents } from './modal.types';
+
+@Component({
+	tag: 'kv-modal',
+	styleUrls: {
+		night: 'modal.night.scss',
+		light: 'modal.light.scss'
+	},
+	shadow: true
+})
+export class KvModal implements IModalConfig, IModalEvents {
```

Comment:
Should we add a prop `closable` to define if this modal can be closed by clicking on the close button or by clicking outside of the container?

Reply:
Also, add the click outside logic as we do on the `kv-dropdown` component.

Reply:
The modal has the prop 'showCloseButton' which may not be the clearest name, I'll try and change that ;)

When the 'hasBackdrop' flag is enabled, only clicks outside of the modal (as they are adjacent divs) will be dispatched as a close event emitted by `clickClose`. Are you recommending a separation of behaviours or should it be kept this way?

Reply:
I'll be adding the "closable" property to the interface as we discussed and control the emission of close events using that flag, thanks!

Reply:
If there's a preference for the outside click detection like the one on the `kv-dropdown` component please let me know, it's a small change :P